extensions/ql-vscode/src/run-remote-query.ts

```diff
@@ -47,6 +47,52 @@ export async function findLanguage(
   return language;
 }
 
+interface RepoListQuickPickItem extends QuickPickItem {
+  repoList: string[];
+}
+
+/**
+ * Gets the repositories to run the query against.
+ */
+async function getRepositories(): Promise<string[] | undefined> {
+  const repoLists = getRemoteRepositoryLists();
+  if (repoLists && Object.keys(repoLists).length) {
+    const quickPickItems = Object.entries(repoLists).map<RepoListQuickPickItem>(([key, value]) => (
+      {
+        label: key,       // the name of the repository list
+        repoList: value,  // the actual array of repositories
+      }
+    ));
+    const quickpick = await window.showQuickPick<RepoListQuickPickItem>(
+      quickPickItems,
+      {
+        placeHolder: 'Select a repository list. You can define repository lists in the `codeQL.remoteRepositoryLists` setting.',
+        ignoreFocusOut: true,
+      });
+    if (quickpick && quickpick.repoList.length > 0) {
+      void logger.log(`Selected repositories: ${quickpick.repoList}`);
+      return quickpick.repoList;
+    } else {
+      void showAndLogErrorMessage('No repositories selected.');
+      return;
+    }
+  } else {
+    void logger.log('No repository lists defined. Displaying text input box.');
+    const remoteRepo = await window.showInputBox({
+      title: 'Enter a GitHub repository in the format <owner>/<repo> (e.g. github/codeql)',
+      placeHolder: '<owner>/<repo>',
+      prompt: 'Tip: you can save frequently used repositories in the `codeql.remoteRepositoryLists` setting',
+      ignoreFocusOut: true,
+    });
+    if (!remoteRepo) {
```

Comment:
Should we also do some basic validation here to ensure the format is indeed `<owner>/<repo>` and fail early (and with a useful error message) if this is indeed the case?

Reply:
Thanks for the help on Slack ✨ Cross-posting the regex you shared here for reference: https://regex101.com/r/krDHm5/2

Reply:
No problem! For future reference, I would also add a comment documenting what the regex is matching, i.e. strings of the form `owner/repo` where:
- `owner` is made up of alphanumeric characters or _single_ hyphens, starting and ending in an alphanumeric character
- `repo` is made up of alphanumeric characters, hyphens, or underscores